src/queries/labelText.ts

```diff
@@ -11,14 +14,38 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+const matchAccessibilityLabelledByText = (
+  rootInstance: ReactTestInstance,
+  nativeID: string | undefined,
+  text: TextMatch
+) => {
+  if (!nativeID) {
+    return false;
+  }
+
+  return (
+    rootInstance.findAll(
+      (node) =>
+        filterNodeByType(node, Text) &&
+        matchStringProp(node.props.nativeID, nativeID) &&
```

Comment:
We should be able to match `<View nativeID="input"><Text>A label</Text></View>`, and I believe this implementation won't work.

Ideally we should do something like `rootInstance.findAll(node => matchStringProp(node.props.nativeID, nativeID)).findAll(node => filterNodeByType(node, Text) && getNodeByText(node, text))`.

And add the relevant test covering that.

Reply:
Hey, I've changed implementation and moved the test to a separate test case. Is it okay?